Assets/Mirage/Runtime/ServerObjectManager.cs

```diff
@@ -236,20 +236,18 @@ public bool AddCharacter(INetworkPlayer player, GameObject character, Guid asset
         /// <param name="client">Client associated to the player.</param>
         /// <param name="character">Player object spawned for the player.</param>
         /// <returns></returns>
-        public bool AddCharacter(INetworkPlayer player, GameObject character)
+        public void AddCharacter(INetworkPlayer player, GameObject character)
         {
             NetworkIdentity identity = character.GetComponent<NetworkIdentity>();
             if (identity is null)
             {
-                logger.Log("AddPlayer: playerGameObject has no NetworkIdentity. Please add a NetworkIdentity to " + character);
-                return false;
+                throw new InvalidOperationException("AddPlayer: playerGameObject has no NetworkIdentity. Please add a NetworkIdentity to " + character);
             }
 
             // cannot have a player object in "Add" version
             if (player.Identity != null)
             {
-                logger.Log("AddPlayer: player object already exists");
-                return false;
+                throw new InvalidOperationException("AddPlayer: player object already exists");
```

Comment:
should be ArgumentException as well, see previous comment

Reply:
fixed